src/screens/Topics/index.js

```diff
@@ -77,8 +76,13 @@ const Topics = () => {
       const index = copytopicSelected.findIndex((data) => data === val);
       if (index > -1) {
         copytopicSelected = copytopicSelected.filter((data) => data !== val);
+        AnalyticsEventTracking.eventTrack(BetterSocialEventTracking.ONBOARDING_TOPICS_UNSELECTED);
       } else {
         copytopicSelected.push(val);
+        AnalyticsEventTracking.eventTrack(BetterSocialEventTracking.ONBOARDING_TOPICS_SELECTED, {
+          ...topic,
+          categoryIndex
+        });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of analytics tracking events when topics are selected or unselected is a good improvement for understanding user behavior. However, it's important to ensure that the `AnalyticsEventTracking.eventTrack` function can handle the case where `topic` or `categoryIndex` are undefined or not in the expected format.



<!-- This is an auto-generated comment by OSS CodeRabbit -->